lib/manager/npm/post-update/npm.ts

```diff
@@ -44,7 +49,7 @@ export async function generateLockFile(
 
     const tagConstraint = await getNodeConstraint(config);
     const execOptions: ExecOptions = {
-      cwd,
+      cwd: lockFileDir,
```

Comment:
I don't think this is right. `cwd` is meant to be an absolute path but you're passing a local path. See https://github.com/renovatebot/renovate/blob/550ab89ba36091017b7c01810ed46f8f1ecf00f2/lib/util/exec/index.ts#L55-L61

Reply:
I think I should prepend `localDir` and pass as `cwd`?

Reply:
What are other managers doing?

Reply:
Pass a local/relative filename as `cwdFile`. We would like to avoid passing an absolute `cwd` from managers because we don't want them needing to know about directory structure